File: Convert_String_LowerCase/Convert.go
/*
    Author : Pakshal Shashikant Jain 
    Date : 12/08/2023
    Program : Write a program which accept string from user and convert it into
              lower case.

              Input : "Marvellous Multi OS"
              Output : marvellous multi os 
*/

//Required Libraries
package main 
import (
	"fmt"
	"bufio"
	"os"
)

//Function To Convert String 
func Convert(arr []byte) []byte {
	var i int = 0;

	//Logic To Convert String 
	for i != len(arr) - 1 {
		if ((arr[i] >= 'A')&&(arr[i] <= 'Z')) {
			arr[i] = arr[i] + 32;
		}		
		
		i++;
	}

	return arr;
}

////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Main Function
func main() {
	var str string = " ";
	var arr []byte;
	var sret []byte;
	var err error = nil;

	//Creating object of bufio Class To Take Input from user
	var bobj = bufio.NewReader(os.Stdin);

	fmt.Println("Jay Ganesh....");	

	//Taking Input from user
	fmt.Println("Enter One String");
	str,err = bobj.ReadString('\n');

	if (err != nil) {
		return;
	}

	//Converting String to array 
	arr = []byte(str);

	//Call To Convert Function
	sret = Convert(arr);

	//Printing Converted String 
 	fmt.Printf("Converted String is : %s",sret);
}

File: Convert_String_LowerCase/Convert.py
"""
    Author : Pakshal Shashikant Jain 
    Date : 12/08/2023
    Program : Write a program which accept string from user and convert it into
              lower case.

              Input : "Marvellous Multi OS"
              Output : marvellous multi os 
"""

#Class Declaration
class ConvertString :
    #Characteristics of Class
    str = " ";
    arr = [];

    #Constructor of Class
    def __init__(self) :
        self.str = " ";
        self.arr = [];

    #Behaviour of Class To Take Input from user
    def Accept(self) :
        #Taking Input from user
        print("Enter One String");
        self.str = input();
        self.arr = list(self.str);

    #Behaviour of Class To Convert String To Lower Case
    def Convert(self) :
        i = 0;
        ascii = 0;

        if(self.arr == None) :
            return;
        
        #Logic 
        while(i != len(self.arr) - 1) :
            if((self.arr[i] >= 'A')and(self.arr[i] <= 'Z')) :
                ascii = ord(self.arr[i]) + 32;
                self.arr[i] = chr(ascii);
            i = i + 1;

####################################################################################################################################################################

#Main Function
def main() :
    print("Jay Ganesh....");

    #Creating object ConvertString Class
    cobj = ConvertString();

    #Call To Behaviour of Class
    cobj.Accept();
    cobj.Convert();

    #Printing Converted String
    print("Converted String is :","".join(cobj.arr));

#Call Back Method To Call Main Function
if __name__ == "__main__" :
    #Call To Main Function
    main();